icx-proxy/src/config/canister_dns_config.rs

```diff
@@ -0,0 +1,150 @@
+use crate::config::dns_alias::DnsAlias;
+use ic_types::Principal;
+
+/// Configuration for determination of Domain Name to Principal
+#[derive(Clone, Debug)]
+pub struct CanisterDnsConfig {
+    dns_aliases: Vec<DnsAlias>,
+}
+
+impl CanisterDnsConfig {
+    /// Create a CanisterDnsConfig instance from command-line configuration.
+    /// dns_aliases: 0 or more entries of the form of dns.alias:canister-id
+    pub fn new(dns_aliases: &[String]) -> anyhow::Result<CanisterDnsConfig> {
+        let dns_aliases = dns_aliases
+            .iter()
+            .map(|alias| DnsAlias::new(alias))
+            .collect::<anyhow::Result<Vec<DnsAlias>>>()?;
+        Ok(CanisterDnsConfig { dns_aliases })
```

Comment:
I filed https://github.com/dfinity/agent-rs/issues/153 which is non blocking for this but would solve a gotcha when passing subdomains as aliases.